packages/backend/src/managers/catalogManager.ts

```diff
@@ -51,31 +53,88 @@ export class CatalogManager extends Publisher<ApplicationCatalog> implements Dis
     this.#disposables = [];
   }
 
+  /**
+   * The init method will start a watcher on the user catalog.json
+   */
   init(): void {
     // Creating a json watcher
-    const jsonWatcher: JsonWatcher<ApplicationCatalog> = new JsonWatcher(
-      path.resolve(this.appUserDirectory, 'catalog.json'),
-      defaultCatalog,
-    );
+    const jsonWatcher: JsonWatcher<ApplicationCatalog> = new JsonWatcher(this.getUserCatalogPath(), {
+      recipes: [],
+      models: [],
+      categories: [],
+    });
     jsonWatcher.onContentUpdated(content => this.onCatalogUpdated(content));
     jsonWatcher.init();
 
     this.#disposables.push(jsonWatcher);
   }
 
+  private loadDefaultCatalog(): void {
+    this.catalog = defaultCatalog;
+    this.notify();
+  }
+
   private onCatalogUpdated(content: ApplicationCatalog): void {
-    // when reading the content on the catalog, the creation is just a string and we need to convert it back to a Date object
-    content.models
-      .filter(m => m.file?.creation)
-      .forEach(m => {
-        if (m.file?.creation) {
-          m.file.creation = new Date(m.file.creation);
+    if (typeof content !== 'object' || !('models' in content) || typeof content.models !== 'object') {
+      this.loadDefaultCatalog();
+      return;
+    }
+
+    const sanitize = this.sanitize(content);
+    this.catalog = {
+      models: [...defaultCatalog.models, ...sanitize.models],
```

Comment:
He can't. What would be the use case ?

Reply:
I think we should check if there are multiple items with the same id. In the rare case that it may happen we could have issues as we manage the models as a map.

Reply:
If that's the goal it already works this way. If you have in the user catalog an entry with an id = one item in the default catalog you will see the custom one